File: JULY-2025/22-07-2025/p1695.py
class Solution:
    def maximumUniqueSubarray(self, nums: List[int]) -> int:
        dic = defaultdict(int)
        subsum = 0
        maxsum = -float('inf')
        l = 0
        for num in nums:
            dic[num] += 1
            subsum += num

            while dic[num] > 1:
                dic[nums[l]] -= 1
                subsum -= nums[l]
                l += 1
            maxsum = max(maxsum, subsum)
        return maxsum

File: JULY-2025/22-07-2025/p1695.c
#define maxFreq 10001
int maximumUniqueSubarray(int* nums, int numsSize) {
    int* freq = calloc(maxFreq, sizeof(int));
    int sum = 0;    // Current Sum
    int maxSum = 0; // Max Sum
    int tail = 0, head = 0;
    for (head = 0; head < numsSize; head++) {
        sum += nums[head];
        freq[nums[head]]++;

        if (freq[nums[head]] != 0) {
            while (freq[nums[head]] > 1) {
                sum -= nums[tail];
                freq[nums[tail]]--;
                tail++;
            }
        }
        if (sum > maxSum) {
            maxSum = sum;
        }
    }
    free(freq);
    return maxSum;
}